cirq-core/cirq/protocols/resolve_parameters_test.py

```diff
@@ -35,16 +35,16 @@ def __init__(self, var):
             self.parameter = var
 
         def _is_parameterized_(self) -> bool:
-            return self.parameter == 0
+            return self.parameter != 0
```

Comment:
Just curious, why this change?

Reply:
Because in the existing test, the `_is_paramaterized_` returns `True` when the class is actually not paramaterized (i.e. `self.parameter is 0` and there's nothing to resolve). 

As a result `cirq.is_parameterized(val)` returns False when `val` is actually parameterized and therefore the output of `cirq.resolve_parameters(val)` changes. 

So this change fixes the test to make the behavior of `_is_paramaterized_` and `resolve_parameters` consistent.